File: src/rails/util/GameIOData.java
package rails.util;

import java.util.List;

import rails.common.GameData;
import rails.game.action.PossibleAction;

/**
 * Combines all elements for gameIO
 */
class GameIOData {

    private GameData gameData;
    private String version;
    private String date;
    private long fileVersionID;
    private List<PossibleAction> actions;
    
    GameIOData(GameData gameData, String version, String date, Long fileVersionID, List<PossibleAction> actions) {
        this.gameData = gameData;
        this.version = version;
        this.date = date;
        this.fileVersionID = fileVersionID;
        this.actions = actions;
    }
    
    GameIOData() {}
    
    
    void setGameData(GameData gameData) {
        this.gameData = gameData;
    }
    
    GameData getGameData() {
        return gameData;
    }
    
    void setVersion(String version) {
        this.version = version;
    }
    
    String getVersion() {
        return version;
    }
    
    void setDate(String date) {
        this.date = date;
    }
    
    String getDate() {
        return date;
    }
    
    void setFileVersionID(long fileVersionID) {
        this.fileVersionID = fileVersionID;
    }
    
    long getFileVersionID() {
        return fileVersionID;
    }
    
    void setActions(List<PossibleAction> actions) {
        this.actions = actions;
    }

    List<PossibleAction> getActions() {
        return actions;
    }
    
    String metaDataAsText() {
        StringBuilder s = new StringBuilder();
        s.append("Rails saveVersion = " + version + "\n");
        s.append("File was saved at " + date + "\n");
        s.append("Saved versionID=" + fileVersionID + "\n");
        s.append("Save game=" + gameData.getGameName() + "\n");
        return s.toString();
    }
    
    String gameOptionsAsText() {
        StringBuilder s = new StringBuilder();
        for (String key : gameData.getGameOptions().keySet()) {
            s.append("Option "+key+"="+gameData.getGameOptions().get(key)+ "\n");
        }
        return s.toString();
    }
    
    String playerNamesAsText() {
        StringBuilder s = new StringBuilder();
        int i=1;
        for (String player : gameData.getPlayers()) {
            s.append("Player "+(i++)+": "+player + "\n");
        }
        return s.toString();
    }
    
}

File: dotnet/RailsLib.Net/Net/Util/GameIOData.cs
using GameLib.Net.Common;
using GameLib.Rails.Game.Action;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

/**
 * Combines all elements for gameIO
 */
namespace GameLib.Net.Util
{
    [JsonObject(MemberSerialization.Fields)]
    public class GameIOData
    {
        private GameData gameData;
        private string version;
        private string date;
        private long fileVersionID;
        [JsonIgnore]
        private List<PossibleAction> actions;

        public GameIOData(GameData gameData, string version, string date, int fileVersionID, List<PossibleAction> actions)
        {
            this.gameData = gameData;
            this.version = version;
            this.date = date;
            this.fileVersionID = fileVersionID;
            this.actions = actions;
        }

        public GameIOData() { }


        public GameData GameData
        {
            get
            {
            return this.gameData;
            }

            set
            {
            this.gameData = value;
            }
        }

        //GameData getGameData()
        //{
        //    return gameData;
        //}

        //public void setVersion(string version)
        //{
        //    this.version = version;
        //}

        public string Version
        {
            get
            {
                return version;
            }
            set
            {
                version = value;
            }
        }

        //void setDate(string date)
        //{
        //    this.date = date;
        //}

        public string Date
        {
            get
            {
                return date;
            }
            set
            {
                date = value;
            }
        }

        //void setFileVersionID(long fileVersionID)
        //{
        //    this.fileVersionID = fileVersionID;
        //}

        public long FileVersionID
        {
            get
            {
                return fileVersionID;
            }
            set
            {
                fileVersionID = value;
            }
        }

        void SetActions(List<PossibleAction> actions)
        {
            this.actions = actions;
        }

        public List<PossibleAction> Actions
        {
            get
            {
                return actions;
            }
            set
            {
                actions = value;
            }
        }

        public string MetaDataAsText()
        {
            StringBuilder s = new StringBuilder();
            s.Append("Rails saveVersion = " + version + "\n");
            s.Append("File was saved at " + date + "\n");
            s.Append("Saved versionID=" + fileVersionID + "\n");
            s.Append("Save game=" + gameData.GameName + "\n");
            return s.ToString();
        }

        public string GameOptionsAsText()
        {
            StringBuilder s = new StringBuilder();
            foreach (string key in gameData.GameOptions.GetOptions().Keys)
            {
                s.Append("Option " + key + "=" + gameData.GameOptions.Get(key) + "\n");
            }
            return s.ToString();
        }

        public string PlayerNamesAsText()
        {
            StringBuilder s = new StringBuilder();
            int i = 1;
            foreach (string player in gameData.Players)
            {
                s.Append("Player " + (i++) + ": " + player + "\n");
            }
            return s.ToString();
        }
    }
}
